pkg/cmd/pipeline/start_test.go

```diff
@@ -0,0 +1,299 @@
+// Copyright © 2019 The Tekton Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package pipeline
+
+import (
+	"fmt"
+	"reflect"
+	"testing"
+
+	tu "github.com/tektoncd/cli/pkg/test"
+	"github.com/tektoncd/pipeline/pkg/apis/pipeline/v1alpha1"
+	"github.com/tektoncd/pipeline/test"
+	tb "github.com/tektoncd/pipeline/test/builder"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	k8stest "k8s.io/client-go/testing"
+)
+
+func Test_start_has_pipeline_arg(t *testing.T) {
+	c := Command(&tu.Params{})
+
+	_, err := tu.ExecuteCommand(c, "start", "-n", "ns")
+
+	if err == nil {
+		t.Error("Expecting an error but it's empty")
+	}
+}
+
+func Test_start_pipeline_not_found(t *testing.T) {
+	ps := []*v1alpha1.Pipeline{
+		tb.Pipeline("test-pipeline", "foo",
+			tb.PipelineSpec(
+				tb.PipelineDeclaredResource("git-repo", "git"),
+				tb.PipelineParam("pipeline-param", tb.PipelineParamDefault("somethingdifferent")),
+				tb.PipelineParam("rev-param", tb.PipelineParamDefault("revision")),
+				tb.PipelineTask("unit-test-1", "unit-test-task",
+					tb.PipelineTaskInputResource("workspace", "git-repo"),
+					tb.PipelineTaskOutputResource("image-to-use", "best-image"),
+					tb.PipelineTaskOutputResource("workspace", "git-repo"),
+				),
+			),
+		),
+	}
+
+	cs, _ := test.SeedTestData(test.Data{Pipelines: ps})
+	p := &tu.Params{Tekton: cs.Pipeline, Kube: cs.Kube}
+
+	pipeline := Command(p)
+	got, _ := tu.ExecuteCommand(pipeline, "start", "test-pipeline-2", "-n", "ns")
+	expected := "Error: " + errInvalidPipeline.Error() + "\n"
+	tu.AssertOutput(t, expected, got)
+}
+
+func Test_start_trigger(t *testing.T) {
+
+	pipelineName := "test-pipeline"
+
+	ps := []*v1alpha1.Pipeline{
+		tb.Pipeline(pipelineName, "foo",
+			tb.PipelineSpec(
+				tb.PipelineDeclaredResource("git-repo", "git"),
+				tb.PipelineParam("pipeline-param", tb.PipelineParamDefault("somethingdifferent")),
+				tb.PipelineParam("rev-param", tb.PipelineParamDefault("revision")),
+				tb.PipelineTask("unit-test-1", "unit-test-task",
+					tb.PipelineTaskInputResource("workspace", "git-repo"),
+					tb.PipelineTaskOutputResource("image-to-use", "best-image"),
+					tb.PipelineTaskOutputResource("workspace", "git-repo"),
+				),
+			), // spec
```

Comment:
need those comments? :)

Reply:
Not necessary but makes it easier to read I think.